conductr_cli/sandbox_proxy.py

```diff
@@ -0,0 +1,107 @@
+from conductr_cli import conduct_main, terminal
+from conductr_cli.constants import DEFAULT_SANDBOX_PROXY_DIR, DEFAULT_SANDBOX_PROXY_CONTAINER_NAME
+from conductr_cli.screen_utils import headline
+import logging
+import pathlib
+import os
+
+
+HAPROXY_VERSION = '1.5'
+HAPROXY_DOCKER_IMAGE = 'haproxy:{}'.format(HAPROXY_VERSION)
+
+HAPROXY_CFG_DIR = '{}/haproxy'.format(DEFAULT_SANDBOX_PROXY_DIR)
+HAPROXY_CFG_PATH = '{}/haproxy.cfg'.format(HAPROXY_CFG_DIR)
+DEFAULT_PROXY_PORTS = [80, 443, 9000]  # These are the ports which will be opened by default to the proxy.
+
+# This is the default config which will be supplied so HAProxy instance within Docker can be started.
+DEFAULT_HAPROXY_CFG_ENTRIES = 'defaults\n' \
+                              '  log global\n' \
+                              '  mode    http\n' \
+                              '  option  httplog\n' \
+                              '  option  dontlognull\n' \
+                              '  timeout connect 5000\n' \
+                              '  timeout client  50000\n' \
+                              '  timeout server  50000\n' \
+                              '\n' \
+                              'frontend conductr-haproxy-test\n' \
+                              '  bind :65535\n' \
+                              '  mode http\n' \
+                              '  monitor-uri /test\n' \
+                              ''
+
+
+def start_proxy(proxy_bind_addr, proxy_ports):
+    validate_docker_present()
+    setup_haproxy_dirs()
```

Comment:
`validate_docker_vm` exit with 1 if Docker is not installed. What we want to do here instead is when docker is not installed, then we skip starting HAProxy and `conductr-haproxy`. So we don't want to exit with error code 1 here. We only want to skip the proxy related steps.

Something like this:

```
if validate_docker_present():
    setup_haproxy_dirs()
    stop_proxy()
    start_docker_instance(proxy_bind_addr, proxy_ports)
    start_conductr_haproxy()
```

To avoid creating an additional docker validation method we could change the `validate_docker_vm` method so that it throws an exception instead of logging the error and exiting with 1. Each caller can then decide how to handle the exceptions (Felix: Basically what you already suggested)